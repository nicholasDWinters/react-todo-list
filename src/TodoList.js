import React, { useState } from 'react';
import Todo from './Todo';
import NewTodoForm from './NewTodoForm';
import { v4 as uuid } from 'uuid';

const TodoList = () => {
    const [todos, setTodos] = useState([]);


    const addTodo = (todo) => {
        setTodos(todos => [...todos, { ...todo, id: uuid() }]);
    }

    const removeTodo = (id) => {
        setTodos(todos.filter(todo => todo.id !== id))
    }

    const editTodo = (todo) => {
        const found = todos.find(t => t.id === todo.id);
        found.task = todo.task;
        setTodos(todos => [...todos]);
    }

    return (
        <div className='TodoList'>
            <NewTodoForm addTodo={addTodo} />
            {todos.map(todo =>

                <Todo task={todo.task} key={todo.id} id={todo.id} removeTodo={removeTodo}
                    editTodo={editTodo} />

            )}
        </div>
    )
};

export default TodoList;