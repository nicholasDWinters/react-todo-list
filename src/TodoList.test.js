import React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import TodoList from './TodoList';

it("renders without crashing", function () {
    render(<TodoList />);
});

it("matches snapshot", function () {
    const { asFragment } = render(<TodoList />);
    expect(asFragment()).toMatchSnapshot();
});

it("should add new todo", function () {
    const { queryByText, getByLabelText } = render(<TodoList />);
    const input = getByLabelText("Todo");
    const btn = queryByText("Add Todo");
    expect(queryByText('shop')).not.toBeInTheDocument();
    fireEvent.change(input, { target: { value: 'shop' } });
    fireEvent.click(btn);
    expect(queryByText('shop')).toBeInTheDocument();
})

//not sure how to get this test to pass ????
it("should allow user to edit todo", function () {
    const { queryByText, getByLabelText } = render(<TodoList />);
    const input = getByLabelText("Todo");
    const btn = queryByText("Add Todo");
    fireEvent.change(input, { target: { value: 'shop' } });
    fireEvent.click(btn);
    fireEvent.click(screen.getByRole('button', { name: 'Edit' }));
    const editInput = getByLabelText("Edit");
    const editBtn = queryByText("Edit Todo");
    fireEvent.change(editInput, { target: { value: 'golfing' } });
    fireEvent.click(editBtn);
    expect(queryByText('shop')).not.toBeInTheDocument();
    expect(queryByText('golfing')).toBeInTheDocument();
})
