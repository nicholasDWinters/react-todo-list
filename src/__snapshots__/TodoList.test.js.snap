// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="TodoList"
  >
    <form
      class="NewTodoForm"
    >
      <label
        class="NewTodoForm-lable"
        for="todo"
      >
        Todo
      </label>
      <input
        class="NewTodoForm-input"
        id="todo"
        name="task"
        placeholder="New Todo"
        type="text"
        value=""
      />
      <button
        class="NewTodoForm-button"
      >
        Add Todo
      </button>
    </form>
  </div>
</DocumentFragment>
`;
